### ISSUE TITLE
Changes on objects within array are not saved

### ISSUE THREAD

**Issue by dnish** [pinqy520/mobx-persist#17]
Hey,
I'm having a persist array like this:

    @persist('list') @observable failedUploads = [];

In some of my functions, I'm changing the values of an object this way:

    this.failedUploads[index].fails++;

The changes were made successfully, I can validate it by console log. But after reopening the app, they get not restored. I still get the initial value (in my case 1) of the property "fails".

**Comment by pinqy520** [comment]
Thanks for report! It fixes in `0.3.2`.

**Comment by dnish** [comment]
@pinqy520 Thank you for the update, but I get the following exception after upgrading to the latest version:

```
[mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Reaction@4] TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at _walk (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=true&minify=false:77294:39)
    at http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=true&minify=false:77295:16
    at Array.map (native)
    at _walk (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=true&minify=false:77294:47)
    at http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=true&minify=false:77295:16
    at Array.map (native)
    at _walk (http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=true&minify=false:77294:47)
    at http://localhost:8081/index.android.bundle?platform=android&dev=true&hot=true&minify=false:77295:16
    at Array.map (native)
```

**Comment by pinqy520** [comment]
@dnish fix in `0.3.3`.....
